db-engine/src/main/java/com/splicemachine/db/catalog/types/IndexDescriptorImpl.java

```diff
@@ -259,15 +259,15 @@ public String indexType()
 	/** @see IndexDescriptor#isAscending */
 	public boolean isAscending(Integer keyColumnPosition) {
 		int i = keyColumnPosition - 1;
-		if (i < 0 || i >= baseColumnPositions.length)
+		if (i < 0 || i >= isAscending.length)
 			return false;
```

Comment:
The code looks good.  I'm just wondering in the base code why both isAscending and isDescending return false for an out-of-range column.  I would assume that since ascending is the default, an out-of-range column should return true (unless we want to catch the out-of-range condition and throw an exception.

Reply:
Indeed, that's a good point. I'll take a look at the call sites of these two functions and file a Jira if there is no special logic relying on the false return values. Probably we should throw than returning a default to prevent subtle bugs.